EIPS/eip-overlay_tree.md

```diff
@@ -0,0 +1,108 @@
+---
+eip: <to be assigned>
+title: Trie format transition with overlay trees
+author: Guillaume Ballet (@gballet)
+discussions-to: https://ethresear.ch/t/overlay-method-for-hex-bin-tree-conversion/7104
+status: Draft
+type: Standards Track
+category: Core
+created: 2020-04-03
+---
+
+## Simple Summary
+
+This EIP proposes a method to convert the state trie format from hexary to binary: new values are directly stored in a binary trie “laid over” the hexary trie. Meanwhile, the hexary trie is converted to a binary trie in the background. When the process is finished, both layers are merged.
+
+## Abstract
+
+This EIP describes a four phase process to complete the conversion.
+
+  * In the first phase, all new state writes are made to an overlay binary trie, while the hexary trie is being converted to binary. The block format is changed to have two storage roots: the root of the nhexary trie and the root of the binary trie.
```

Comment:
If the overlay is hexary, then it would have to be converted during the phase 3 merge. Having an overlay binary trie allows the clients to directly switch to binary tries. Then it gives clients time to perform the conversion in the background.